tests/unit/pywbemcli/simple_foo_mock_script.py

```diff
@@ -0,0 +1,100 @@
+"""
+Test mock script that prepares an Interop namespace and namespace provider,
+and then compiles MOF that has a pragma namespace 'foo'.
+
+This mock script uses 'new-style' setup for Python >=3.5 and 'old-style' setup
+otherwise, and is therefore useable for all supported Python versions.
+On Python <3.5, the tests using this script must be run from the repo main
+directory.
+"""
+
+import sys
+import os
+import six
+import pywbem_mock
+
+# interop namespace used by this mock environment
+INTEROP_NAMESPACE = 'interop'
+
+
+def register_dependents(conn, this_file_path, dependent_file_names):
+    """
+    Register a dependent file name with the pywbemcli dependent file api.
+    This insures that any change to a dependent file will cause the
+    script to be recompiled.
+    """
+    if isinstance(dependent_file_names, six.string_types):
+        dependent_file_names = [dependent_file_names]
+
+    for fn in dependent_file_names:
+        dep_path = os.path.join(os.path.dirname(this_file_path), fn)
+        conn.provider_dependent_registry.add_dependents(this_file_path,
+                                                        dep_path)
+
+
+def _setup(conn, server, verbose):
+    # pylint: disable=unused-argument
+    """
+    Setup for this mock script.
+
+    Parameters:
+      conn (FakedWBEMConnection): Connection
+      server (PywbemServer): Server
+      verbose (bool): Verbose flag
+    """
+
+    if sys.version_info >= (3, 5):
+        this_file_path = __file__
+    else:
+        # Unfortunately, it does not seem to be possible to find the file path
+        # of the current script when it is executed using exec(), so we hard
+        # code the file path. This requires that the tests are run from the
+        # repo main directory.
+        this_file_path = 'tests/unit/pywbemcli/simple_foo_mock_script.py'
+        assert os.path.exists(this_file_path)
+
+    # Prepare an Interop namespace and namespace provider
+
+    interop_mof_file = 'mock_interop.mof'
```

Comment:
came comment about packaging the interop namespace setup into pywbemcli or even some of it into pywbem

Reply:
see above